### PR TITLE
Change all occurrences to use Intl.ListFormat

### PR DESCRIPTION
**Pull request by silamon**
## Proposed change
Finish off #14589 for other triggers and conditions after #16857 got merged in. This should help at translating the descriptions of triggers, conditions and actions in the future. For now, it's hardcoded as English.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #14589 
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/data/automation_i18n.ts [hunk 1]
```diff
@@ -268,11 +258,9 @@ export const describeTrigger = (
           base += " to any state";
         }
       } else if (Array.isArray(trigger.to)) {
-        let to = "";
-        for (const [index, state] of trigger.to.entries()) {
-          to += `${index > 0 ? "," : ""} ${
-            trigger.to.length > 1 && index === trigger.to.length - 1 ? "or" : ""
-          } '${
+        const to: string[] = [];
+        for (const [_, state] of trigger.to.entries()) {
```
Comment (bramkragten) [inline src/data/automation_i18n.ts:262]:
```suggestion
        for (const state of trigger.to.values()) {
```

File: src/data/automation_i18n.ts [hunk 2]
```diff
@@ -208,13 +201,9 @@ export const describeTrigger = (
...
       } else if (Array.isArray(trigger.from)) {
-        let from = "";
-        for (const [index, state] of trigger.from.entries()) {
-          from += `${index > 0 ? "," : ""} ${
-            trigger.from.length > 1 && index === trigger.from.length - 1
-              ? "or"
-              : ""
-          } '${
+        const from: string[] = [];
+        for (const [_, state] of trigger.from.entries()) {
```
Comment (bramkragten) [inline src/data/automation_i18n.ts:205]:
```suggestion
        for (const state of trigger.from.values()) {
```

File: src/data/automation_i18n.ts [hunk 3]
```diff
@@ -515,9 +504,9 @@ export const describeTrigger = (
     }
 
     if (Array.isArray(trigger.zone)) {
-      for (const [zone] of trigger.zone.entries()) {
+      for (const [_, zone] of trigger.zone.entries()) {
```
Comment (bramkragten) [inline src/data/automation_i18n.ts:507]: same

File: src/data/automation_i18n.ts [hunk 4]
```diff
@@ -501,9 +490,9 @@ export const describeTrigger = (
     const states = hass.states;
 
     if (Array.isArray(trigger.entity_id)) {
-      for (const [entity] of trigger.entity_id.entries()) {
+      for (const [_, entity] of trigger.entity_id.entries()) {
```
Comment (bramkragten) [inline src/data/automation_i18n.ts:493]: same

File: src/data/automation_i18n.ts [hunk 5]
```diff
@@ -537,47 +526,39 @@ export const describeTrigger = (
...
     const states = hass.states;
 
     if (Array.isArray(trigger.source)) {
-      for (const [index, source] of trigger.source.entries()) {
-        sources += `${index > 0 ? "," : ""} ${
-          trigger.source.length > 1 && index === trigger.source.length - 1
-            ? "or"
-            : ""
-        } ${source}`;
+      for (const [_, source] of trigger.source.entries()) {
```
Comment (bramkragten) [inline src/data/automation_i18n.ts:534]: same

File: src/data/automation_i18n.ts [hunk 6]
```diff
@@ -708,21 +689,24 @@ export const describeCondition = (
...
+          entities.push(computeStateName(hass.states[entity]) || entity);
         }
       }
-      if (entities) {
-        base += ` ${entities} ${condition.entity_id.length > 1 ? "are" : "is"}`;
+      if (entities.length !== 0) {
+        const conjunctionFormatter = new Intl.ListFormat("en", {
+          style: "long",
+          type: "conjunction",
+        });
```
Comment (bramkragten) [inline src/data/automation_i18n.ts:702]: Can we move this to the top with `disjunctionFormatter`

File: src/data/automation_i18n.ts [hunk 7]
```diff
@@ -735,20 +719,16 @@ export const describeCondition = (
...
           : condition.entity_id
       ];
     if (Array.isArray(condition.state)) {
-      for (const [index, state] of condition.state.entries()) {
-        states += `${index > 0 ? "," : ""} ${
-          condition.state.length > 1 && index === condition.state.length - 1
-            ? "or"
-            : ""
-        } '${
+      for (const [_, state] of condition.state.entries()) {
```
Comment (bramkragten) [inline src/data/automation_i18n.ts:730]: same

File: src/data/automation_i18n.ts [hunk 8]
```diff
@@ -947,9 +928,9 @@ export const describeCondition = (
     const states = hass.states;
 
     if (Array.isArray(condition.entity_id)) {
-      for (const [entity] of condition.entity_id.entries()) {
+      for (const [_, entity] of condition.entity_id.entries()) {
```
Comment (bramkragten) [inline src/data/automation_i18n.ts:931]: same

File: src/data/automation_i18n.ts [hunk 9]
```diff
@@ -961,9 +942,9 @@ export const describeCondition = (
     }
 
     if (Array.isArray(condition.zone)) {
-      for (const [zone] of condition.zone.entries()) {
+      for (const [_, zone] of condition.zone.entries()) {
```
Comment (bramkragten) [inline src/data/automation_i18n.ts:945]: same

**Comment by karwosts** [comment]
Time trigger is still manually building a comma separated `or` list if you wanted to get all of them.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**